tuf_conformance/test_keys.py

```diff
@@ -8,69 +8,73 @@
+    """This is a helper for tests for key thresholds.
+    It sets a threshold of 3 for snapshot metadata,
+    adds three keys, bumps the repo metadata and
+    refreshes the client"""
+
+    # Set treshold to 3 for snapshot metadata
     repo.root.roles[Snapshot.type].threshold = 3
-    repo.bump_root_by_one()  # v2
 
-    # Add a legitimate key:
+    # Add three keys for snapshot:
     repo.add_key(Snapshot.type)
-    repo.update_timestamp()
-    repo.update_snapshot()
-    repo.bump_root_by_one()  # v3
-
-    assert client.refresh(init_data) == 1
-
-    # Add a legitimate key:
     repo.add_key(Snapshot.type)
     repo.add_key(Snapshot.type)
     repo.update_timestamp()
-    repo.update_snapshot()
-    repo.bump_root_by_one()  # v4
+    repo.update_snapshot()  # v2
+    repo.bump_root_by_one()  # v2
 
     assert len(repo.root.roles[Snapshot.type].keyids) == 4
 
     assert client.refresh(init_data) == 0
-    assert client.version(Snapshot.type) == 3
-    assert client.version(Root.type) == 4
+    assert client.version(Snapshot.type) == 2
+    assert client.version(Root.type) == 2
 
 
 def test_root_has_keys_but_not_snapshot(
     client: ClientRunner, server: SimulatorServer
 ) -> None:
-    """This test adds keys to the repo root MD to test for cases
-    where are client might calculate the threshold from only the
-    roots keys and not check that the snapshot MD has the same
-    keys"""
+    """This test adds keys for snapshot metadata only to the
+    repo root MD to test for a case where a client might
+    calculate the threshold from only the key in the root MD
+    and not check that the snapshot MD also has the same keys.
+    For example, the goal is to bring the repository into a state where:
+
+    1. repo.root.roles.snapshot.keyids has the following keyids:
+      1.a: ccce1a69b4eea9daf315d8a9c43fffd8ee541bfb41fdcb773657192af51d3cfa
+      1.b: 4f7c454c55a81918fa1fc1f513d510f36c3efa796711d9f5883602f91a5b9da7
+      1.c: 146e0dc5038d3b8772f5d4bd6626fe1436f656494b501628a2b54d6aa6a6edfe
+      1.d: 3c80449ebd33c67ff8d9befce23c534b50d97ed805e52da8c746e6a1daefab8e
+    2: repo.root.roles.snapshot.threshold is 4
+    3: repo.snapshot.signatures has the following keyids:
+      3.a: ccce1a69b4eea9daf315d8a9c43fffd8ee541bfb41fdcb773657192af51d3cfa
+      3.b: 4f7c454c55a81918fa1fc1f513d510f36c3efa796711d9f5883602f91a5b9da7
+      3.c: 146e0dc5038d3b8772f5d4bd6626fe1436f656494b501628a2b54d6aa6a6edfe
+
+    The test ensures that clients do not update snapshot in this scenario.
+    """
     init_data, repo = server.new_test(client.test_name)
 
     assert client.init_client(init_data) == 0
     assert client.refresh(init_data) == 0
 
     initial_setup_for_key_threshold(client, repo, init_data)
 
-    # Increase the threshold
-    repo.root.roles[Snapshot.type].threshold = 5
-    repo.bump_root_by_one()  # v5
+    # Set the threshold to 4 and remove a signer from snapshot
+    # metadata so that root has 4 keys and snapshot has 3
```

Comment:
nit: this is still a really unclear comment: 
> root has 4 keys and snapshot has 3

snapshot role (in root metadata) contains 4 keys, snapshot metadata is signed by 3 keys

Reply:
Renamed and reworded in https://github.com/theupdateframework/tuf-conformance/pull/160/commits/7b3260ac679a5dbfc78ff1dd1ca7855eaf6ec6c0. LMKWYT.